### ISSUE TITLE
Doc: need documentation on table & range query support 

### ISSUE THREAD

**Issue by frankyclu** [hyperledger-archives/fabric#1504]
## Description

Though we got sample code for table and range query, 

https://github.com/hyperledger/fabric/tree/master/examples/chaincode/go/asset_management
https://github.com/hyperledger/fabric/blob/master/examples/chaincode/go/map/map.go

It hasn't been documented and people are generally not aware of these functionalities 

I got this request from a private message:

> Currently the world state is recorded in the key-value store. But when building a complicated application, key-value store is often not efficient to handle complicated data structure.

**Comment by sachikoy** [comment]
@frankyclu We are aware of range query and table, and using them. However, even using these features, KVS is not very efficient when we have a lot of data, and need to do queries by data values. 
For example, let's say many trade data are recorded in blockchain, and want to get the list of trade that matches certain conditions (e.g., amount, date,etc.). If you implement naively, you need to read each data from table to check the value. This will be quite inefficient and slow if you have millions of trade data in KVS. There are some known workaround to mitigate the overhead, such as building index. However, it is cumbersome, and there are concerns of overhead, e.g., read/write/serialisation. In RDB, we do not need to worry about such issues because of SQL.

**Comment by frankyclu** [comment]
@sachikoy We may need to open another issue to cover the scenario you described. 

One of the reason of using KV store over something like SQLite is due to the performance concerns. 

Chaincode state now supports multiple columns, would allowing range query on each columns satisfy the use case you are trying cover?

@srderson thoughts?

**Comment by srderson** [comment]
@sachikoy Yes, please open a second issue. The more detailed you can be about the types of queries needed, the better.

**Comment by christo4ferris** [comment]
@frankyclu @srderson was the other issue ever opened? Please link this one if so, or open the other and reference. I presume this is still relevant?
